cli/pack/deb_control_dir.go

```diff
@@ -0,0 +1,252 @@
+	log.Debugf("Create DEB control file")
+	err := os.MkdirAll(destDirPath, dirPermissions)
+	if err != nil {
+		return err
+	}
+
+	name := packCtx.Name
+	if packCtx.Name == "" {
+		name = "bundle"
+	}
+	ver := packCtx.Version
+	if packCtx.Version == "" {
+		ver = "0.1.0.0"
+	}
+
+	debControlCtx := map[string]interface{}{
+		"Name":         name,
+		"Version":      ver,
+		"Maintainer":   defaultMaintainer,
+		"Architecture": runtime.GOARCH,
+		"Depends":      "",
+	}
+
+	deps, err := ParseDependencies(packCtx.RpmDeb.Deps)
+	if err != nil {
+		return err
+	}
+
+	if packCtx.RpmDeb.WithTarantoolDeps {
+		ttBinDeps, err := getTntTTVersions(packCtx)
+		if err != nil {
+			return err
+		}
+		deps = append(deps, ttBinDeps...)
+	}
+
+	if len(deps) > 0 {
+		addDependenciesDeb(&debControlCtx, deps)
+	}
+
+	err = createControlFile(destDirPath, &debControlCtx)
+	if err != nil {
+		return err
+	}
+	log.Infof("Created control in %s", destDirPath)
+
+	// add postinst and preinst scripts step
+	if packCtx.RpmDeb.PreInst == "" {
+		pif, err := os.Create(filepath.Join(destDirPath, "preinst"))
+		if err != nil {
+			return err
+		}
+		defer pif.Close()
+		templ, err := template.New("postinst").Parse(PreInstScriptContent)
+		err = templ.Execute(pif, map[string]interface{}{"Name": name})
+		if err != nil {
+			return err
+		}
+	} else {
+		err = copy.Copy(packCtx.RpmDeb.PreInst, filepath.Join(destDirPath, "preinst"))
+		if err != nil {
+			return err
+		}
+	}
+	if packCtx.RpmDeb.PostInst == "" {
+		pif, err := os.Create(filepath.Join(destDirPath, "postinst"))
+		if err != nil {
+			return err
+		}
+		defer pif.Close()
+		templ, err := template.New("postinst").Parse(PostInstScriptContent)
+		err = templ.Execute(pif, map[string]interface{}{"Name": name})
+		if err != nil {
+			return err
+		}
+	} else {
+		err = copy.Copy(packCtx.RpmDeb.PostInst, filepath.Join(destDirPath, "postinst"))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getDebRelation(relation string) string {
+	if relation == ">" || relation == "<" {
+		// Deb format uses >> and << instead of > and <
+		return fmt.Sprintf("%s%s", relation, relation)
+	} else if relation == "==" {
+		return "="
+	}
+
+	return relation
+}
+
+func addDependenciesDeb(debControlCtx *map[string]interface{}, deps PackDependencies) {
+	var depsList []string
+
+	for _, dep := range deps {
+		for _, r := range dep.Relations {
+			depsList = append(depsList, fmt.Sprintf("%s (%s %s)",
+				dep.Name, getDebRelation(r.Relation), r.Version))
+		}
+
+		if len(dep.Relations) == 0 {
+			depsList = append(depsList, dep.Name)
+		}
+	}
+
+	(*debControlCtx)["Depends"] = strings.Join(depsList, ", ")
+}
+
+func createControlFile(basePath string, debControlCtx *map[string]interface{}) error {
+	f, err := os.Create(filepath.Join(basePath, "control"))
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	tmpl, err := template.New("controlFileContent").Parse(controlFileContent)
```

Comment:
Can you use templates API from util package for this? Probably, some minor modifications required.
I see the code duplication for creating a file, rendering a template text to the file.

Reply:
This api is not for this purpose, there is no sense to use it.

Reply:
We need to remove code duplication for text rendering: create file, create new template, parse text, render template to the file. It is repeated 3 times. We already have similar functionality in the templates package. It makes sense to have all templates related stuff there.